File: app/service/get_components.py
# -- coding: utf-8 --
import logging

from config.omc import omc


def GetComponents(class_name, component_name=None):
    Components_data = omc.getComponents(class_name)
    component_data = []
    if Components_data != "Error":
        try:
            for i in Components_data:
                if component_name and i[1] == component_name:
                    component_data = i
                    break
                elif not (i[8] == "parameter" or  i[3] == "protected" or i[4] == "True") and not component_name:
                    component_data.append(i)
        except Exception as e:
            logging.error(e)
    return component_data

File: app/service/GetComponents.go
package service

import "yssim-go/library/omc"

func GetComponents(className string, componentName string) []interface{} {
	componentsData := omc.OMC.GetComponents(className)
	var componentData []interface{}
	for i := 0; i < len(componentsData); i++ {
		cData := componentsData[i].([]interface{})
		switch {
		case componentName != "" && cData[1] == componentName:
			componentData = cData
			break
		case (cData[3] != "protected" || cData[4] != "True" || cData[8] != "parameter") && componentName == "":
			componentData = append(componentData, cData)
		}
	}
	return componentData
}
